src/main/java/gov/cancer/pageobject/cts/advanced_search_page_components/CancerTypeSection.java

```diff
@@ -101,4 +104,20 @@ public AutoSuggestField getCancerTypeCondition() {
   public Button getAllButton() {
     return all;
   }
+
+  /**
+   * Getter for  Cancer type input box
+   * @return
+   */
+  public boolean isCancerTypeInputBoxVisible() {
+    return ElementHelper.findElement(section, CANCER_TYPE_INPUT_LOCATOR).isDisplayed();
```

Comment:
Isn't cancer type input box an AutoSuggest field? Why create a separate method to verify it's visibility when you already have an AutoSuggest object named `cancerTypeCondition` which inherits `isVisible` method from Component?

Reply:
this method is only used here :
` @Test(dataProvider = "getCancerType")
  public void verifyInputBoxOnClickAllButton(String path) {
    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
      page.getCancerTypeSection().scrollUntilAllButtonVisible();
      //Click All button
      page.getCancerTypeSection().getAllButton().click();
      //asserting that input box for the Primary Cancer Type dropdown is displayed
      Assert.assertTrue(page.getCancerTypeSection().getCancerTypeInputBox().isDisplayed(), "Input box is not visible");
    });
  }`

Why not change it to:
`@Test(dataProvider = "getCancerType")
  public void verifyInputBoxOnClickAllButton(String path) {
    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
      page.getCancerTypeSection().scrollUntilAllButtonVisible();
      //Click All button
      page.getCancerTypeSection().getAllButton().click();
      //asserting that input box for the Primary Cancer Type dropdown is displayed
      Assert.assertTrue(page.getCancerTypeSection().getCancerTypeCondition().isVisible(), "Input box is not visible");
    });
  }`

Then you do not need an extra code with

Reply:
done